tests/run-tests.py

```diff
@@ -896,22 +877,93 @@ def run_one_test(test_file):
             skipped_tests.append(test_name)
             return
 
-        testcase_count.add(len(output_expected.splitlines()))
+        # Look at the output of the test to see if unittest was used.
+        uses_unittest = False
+        output_mupy_lines = output_mupy.splitlines()
+        if any(
+            line == b"ImportError: no module named 'unittest'" for line in output_mupy_lines[:3]
```

Comment:
This assumes there's no other output before the test tries to import unittest, which isn't necessarily the case. Minor issue though since the test will still fail.

Reply:
If a test uses `unittest` it's very unlikely it will print anything before `import unittest`.  The pattern I've been using is:
- try-import modules that you need for the test, and SKIP if they don't exist
- import unittest (this is done after checking if the test should be SKIPed so you don't need unittest if you won't anyway run the test)
- possibly define some state, helper functions, etc
- define the main test class(es)
- use the standard `if __name__ == "__main__": unittest.main()` code at the end

That said, I could change the code here to check the last few lines (really it only needs to check the last line) of the output?

Reply:
> If a test uses unittest it's very unlikely it will print anything before import unittest

I agree that it is unlikely, and also it seems to be a bit not done to output during import phase but I do have such tests :) 

> That said, I could change the code here to check the last few lines (really it only needs to check the last line) of the output?

Seems very sensible